qiskit/primitives/primitive_job.py

```diff
@@ -46,19 +48,31 @@ def _submit(self):
         self._future = executor.submit(self._function, *self._args, **self._kwargs)
         executor.shutdown(wait=False)
 
+    def _prepare_dump(self):
+        """This method allows PrimitiveJob to be serialized"""
+        _ = self.result()
+        _ = self.status()
+        self._future = None
+
```

Comment:
When is this function called during serialization? I can't see anywhere where the method is called in the PR itself?

Reply:
Would it be better to make this happen in one of the serialization methods that python has built-in? I believe this could be executed on one of the methods, avoiding that function call. You could include the lines here in one of the reserved methods for serialization: `__reduce__`, `__getstate__`, `__getnewargs__`, depending on your needs.

Reply:
Thank you for your suggestion. I think the following should work (to remove non-pickable object `_future`) by replacing `_prepare_dump` with the following.

```python
    def __getstate__(self):
        _ = self.result()
        _ = self.status()
        state = self.__dict__.copy()
        state["_future"] = None
        return state

    def __setstate__(self, state):
        self.__dict__.update(state)
        self._future = None
```

Reply:
I made a patch to this PR https://github.com/kt474/qiskit-terra/pull/1.